pypesto/problem.py

```diff
@@ -149,15 +148,112 @@ def normalize_input(self):
             raise AssertionError("lb dimension invalid.")
         if self.ub.size != self.dim:
             raise AssertionError("ub dimension invalid.")
-        if self.x_guesses.shape[1] != self.dim:
-            raise AssertionError("x_guesses form invalid.")
+        if check_x_guesses:
+            if self.x_guesses.shape[1] != self.dim:
+                raise AssertionError("x_guesses form invalid.")
         if len(self.x_names) != self.dim_full:
             raise AssertionError("x_names must be of length dim_full.")
         if len(self.x_fixed_indices) != len(self.x_fixed_vals):
             raise AssertionError(
                 "x_fixed_indices and x_fixed_vals musti have the same length."
             )
 
+    def fix_parameters(self, parameter_indices, parameter_vals):
+        """
+        Fix specified parameters to specified values
+        """
+
+        if not parameter_indices is list:
```

Comment:
you mean ``if not isinstance(parameter_indices, list)``?

Reply:
yep...
for some reason, that worked (or was never hit...)